src/lib/process-lock.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * This module contains the functionality for locking & unlocking resources
+ * within the Supervisor Node process, useful for methods that need to acquire
+ * exclusive access to a resource across multiple ticks in the event loop, async
+ * functions for example.
+ *
+ * It is different from lockfile and update-lock modules, which handle
+ * inter-container communication via lockfiles.
+ *
+ * TODO:
+ * - Use a maintained solution such as async-lock
```

Comment:
I personally also liked [BetterLock ](https://www.npmjs.com/package/better-lock) although it would be even better if we could avoid locking